internal/logging/rotate.go

```diff
@@ -49,7 +49,7 @@ func rotateLogs(files []fs.FileInfo, timeCutoff time.Time, amountCutoff int) []f
 func rotateLogsOnDisk() {
 	// Clean up old log files
 	logDir := filepath.Dir(FilePath())
-	files, err := ioutil.ReadDir(logDir)
+	files, err := ioutil.ReadDir(logDir) //nolint:staticcheck
```

Comment:
Whoops, I meant to come back to this one. This is because `os.ReadDir` returns a list of `fs.DirEntry` whereas `ioutil.ReadDir` returns a list of `fs.FileInfo`. For the former, we have to do some extra processing to get the `ModTime` which is used in the `rotateLogs` function.